### ISSUE TITLE
How can I get the logits for one whole sequence in the asr task?

### ISSUE THREAD

**Issue by Thordlos** [tensorflow/lingvo#290]
Refer to the method of obtaining the value of the loss function, which collect the loss value using "tf.add_to_collection" function in "decoder.py/_ComputeMetrics" , and retrieve it in appropriate place via "tf.get_collection".

I try to collect the ASR logits(or say the layer before Cross-Entropy loss function) in the same way, and I get the logits with tensor of shape[batch, time, num_classes], it's turly(5, 140, 16328) in Param of "asr.librispeech.Librispeech960Wpm".

But it's not working completely correct, when I get_collection it and print, it's only working well in time of 0-20, and are padding in time of 20-140. I wonder if it's due to seq2seq task or else, and how to get the logits of asr or lm correctly.

**Comment by drpngx** [comment]
Did you count the labels? The second dimension is probably labels, not
utterance time.

**Comment by drpngx** [comment]
Right. You need the paddings tensor to mask for variable length. Did you
answer your question?
